planner/core/exhaust_physical_plans.go

```diff
@@ -548,15 +548,19 @@ func (p *LogicalJoin) buildIndexJoinInner2TableScan(
 		return nil
 	}
 	keyOff2IdxOff := make([]int, len(innerJoinKeys))
+	newOuterJoinKeys := make([]*expression.Column, 0)
 	pkMatched := false
 	for i, key := range innerJoinKeys {
-		if !key.Equal(nil, pkCol) {
+		if !key.Equal(nil, pkCol) && !outerJoinKeys[i].Equal(nil, pkCol) {
```

Comment:
We don't need to change this line. `pkCol` is from innerTable, it won't be equal to any of the outerJoinKey.